protostar/git/git.py

```diff
@@ -1,52 +1,72 @@
 import subprocess
```

Comment:
Can we use this opportunity to rename this file? Maybe to `ops`/`global_ops`/`commands` ?

Reply:
`ops` doesn't say much. `Commands` can be confused with Protostar commands. I can change it to `git_utils`.

Reply:
i think we don't like `utils` that much as well

Reply:
`GitService`?

Reply:
> They typically sit dormant waiting for other objects to call them, but because they’re often relied upon by so many objects, they may contain responsibilities that relatively stable, not changing often.

> For example, utility classes like TextUtil, DateUtil, NumberUtil provide respective text, date, and number utilities and probably don’t change dramatically (or at all for that matter). For this reason, we’d say they’re stable. Just think about what would happen if we had to change the JSON.stringify method in JavaScript.

Reply:
> Commands can be confused with Protostar commands

this part i don't get, every function here has `git` in its' name, so how can we confuse it?

Reply:
I don't get why we don't like utils that much 🤔

Reply:
@MaksymilianDemitraszek those above are git commands wrappers.
There is always a better name than `utils`, `utils` is just a lazy way out for a module name that you don't have an immediate idea for :P Moreover, it does not provide any information about the functionality of the module

Reply:
This file contains small, git related functions, and often such files are named utils or helpers.

`git_helpers`?

> Commands can be confused with Protostar commands

this part i don't get, every function here has git in its' name, so how can we confuse it?
In this codebase, commands usually refer to Protostar CLI commands. Also, using git CLI through a subprocess is an implementation detail which should be hidden from the outside world. I think we are trying too hard to avoid the u-word. IMO, utils is fine as long as it's prefixed and it's not a directory.

Reply:
Why not git_wrappers?

Reply:
> A descriptive name for this file could be git_interface.py. This name indicates that the file contains a high-level interface for interacting with Git and encapsulates the underlying subprocess calls to Git. The name also conveys the purpose of the file and what it is used for, which can be useful for someone unfamiliar with the code.
~ChadGPT

Reply:
Interesting, but not relevant since it's text-based AI, has no knowledge about our project and our needs/preferences. Thus i would kindly ask to refrain from posting output from ChatGPT as an argument for the discussion.

Reply:
`git_globals_facade`/`git_facade`? i think this one fits nicely

Reply:
> git_globals_facade/git_facade? i think this one fits nicely

I don't know how to interpret globals. I am not sure about the facade.


> Interesting, but not relevant since it's text-based AI, has no knowledge about our project and our needs/preferences. Thus i would kindly ask to refrain from posting output from ChatGPT as an argument for the discussion.

ChatGPT has many flaws, but it's quite good with summarization. It had a knowledge about the file contents and that what matters. It suggested names that we both came up with.

![Screenshot 2023-02-10 at 17 13 01](https://user-images.githubusercontent.com/9629577/218140684-b3cbad41-5809-44f2-a154-074b35a4a73b.png)